src/Cardano/Wallet.hs

```diff
@@ -153,6 +222,15 @@ mkWalletLayer db network = WalletLayer
         putCheckpoint db (PrimaryKey wid) cp'
         unsafeRunExceptT $ putTxHistory db (PrimaryKey wid) txs -- Safe after ^
 
+    readWalletId
+        :: WalletId
+        -> ExceptT ReadWalletError IO (Wallet SeqState)
+    readWalletId wid =
```

Comment:
`readWalletId` --> `readWallet` ?

Reply:
Think I'll call it `readWallet'` to avoid collisions.

Reply:
Collision with the DB ? I went for a qualified import `DB.putCheckpoint` `DB.readWallet` etc .. :thinking:

Reply:
Collision with the `readWallet` field of the record :)

Reply:
Ah :| And what's wrong with re-using the `readWallet` from the wallet layer ?

Reply:
Ah 😮 can you do that? I thought about it, but didn't see how

Reply:
i.e
```haskell
WalletLayer {
readWallet = ...
signTx = readWallet self -- how?
}
```

Reply:
He. Good point. That's _probably_ possible through some Haskell magic I don't _yet_ know. For now, the "simplest" approach I know would be to move the definition of the `readWallet` in the where clause and use that (basically, what you just did, but without duplicating the logic for `readWallet`).